geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedQueryRunner.scala

```diff
@@ -34,15 +36,18 @@ import org.opengis.filter.Filter
   * @param ds merged data store
   * @param stores delegate stores
   */
-class MergedQueryRunner(ds: MergedDataStoreView, stores: Seq[(DataStore, Option[Filter])])
+class MergedQueryRunner(ds: HasGeoMesaStats, stores: Seq[(Queryable, Option[Filter])])
```

Comment:
This is an interesting sharpening....

Reply:
yeah, there was a lot of common code between the LambdaQueryRunner and this class - this abstraction lets us use this one for both, so I don't have to fix all the local/remote arrow calls in both places.

Reply:
Is the Lambda DS just a MergedView DS with a write path?:)

Reply:
kind of... the 'transient' part of it isn't a full-fledged data store, and it also has some logic to persist things that haven't been modified in a while. We could probably abstract it out to where that is the case, which might make it easier to support non-accumulo back-ends.

Reply:
Yeah, that sounds like 1) a lot of work, and 2) a somewhat good idea if the Lambda DS idea 'takes off' and gets used elsewhere.